docs/sources/developers/plugins/extend-the-grafana-ui-with-links.md

```diff
@@ -1,25 +1,78 @@
 ---
-title: Use extensions to add links to app plugins
+title: Use extensions to add new functionality in the Grafana user interface
 description: Learn how to add links to the Grafana user interface from an app plugin
 ---
 
-# Use extensions to add links to app plugins
+# The when, why and how of UI extensions
 
-You can use the Plugin extensions API with your Grafana app plugins to add links to the Grafana UI. This feature lets you send users to your plugin's pages from other spots in the Grafana application.
+Before we go into details we need to cover the two major concepts in the UI extensions feature of Grafana.
+
+1. Extension point - a place in the UI where plugins can provide new functionality to the end user. Examples of extension points are the dashboard panel menu or toolbar in explore.
+2. Extensions - New functionality, usually registered by an app plugin, that will be displayed at an extension point. Examples of extensions are the possibility to create an incident directly from the dashboard panel menu.
+
+![panel menu](https://user-images.githubusercontent.com/172951/242723354-a10d6238-22f1-4458-b85e-ac3c7f014b22.png)
+
+In the example above we have one extension point with two extensions registered. This highlights one of the benefits of using UI extensions. Once you have added an extension point to your UI it can be extended multiple times by multiple plugins.
+
+## When should I add an extension point?
```

Comment:
When and Why do blur a little, partly as a result of when coming before Why. I think if we re-order this will naturally clear up. I can help with this.